fix: enhance v2 tasklist api with processName

konstantinos-mylosis opened the pull request:
## Description

#### Issue
`/v2/user-tasks/search` endpoint did not return field `processName` in the response like `v1/tasks/search` did.

Tasklist UI -> Tasks -> All open tasks
The information retrieved by v2 displayed the `processId` rather than the `processName`.

#### Fix
Use existing following chain and enhance with processName
UserTaskServices -> ProcessCache -> ProcessDefinitionProvider -> ProcessDefinitionServices.search

BEFORE fix
<img width="1315" height="142" alt="image" src="https://github.com/user-attachments/assets/f618c60d-4224-436c-af0e-6124cc4fbebc" />

AFTER fix
<img width="1399" height="193" alt="image" src="https://github.com/user-attachments/assets/86a41caa-de15-443c-bb5e-dbf343759492" />

## Related issues

closes #34961

konstantinos-mylosis committed:
fix: enhance v2 tasklist api with processName

eppdot reviewed (changes requested):
@konstantinos-mylosis looks good overall. I just left a mix of questions, suggestions that I would like to sort out together 😄

konstantinos-mylosis commented on zeebe/gateway-protocol/src/main/proto/rest-api.yaml:
```diff
@@ -5831,6 +5831,9 @@ components:
         elementInstanceKey:
           type: string
           description: The key of the element instance.
+        processName:
```
This endpoint appears to be available only via REST, not through gRPC.

konstantinos-mylosis commented on service/src/main/java/io/camunda/service/cache/ProcessCacheResult.java:
```diff
@@ -23,9 +23,14 @@ public ProcessCacheItem getProcessItem(final Long processDefinitionKey) {
-        Map.of(processDefinitionKey, new ProcessCacheItem(Map.of(elementId, cachedName))));
+        Map.of(
+            processDefinitionKey,
+            new ProcessCacheItem(processName, Map.of(elementId, cachedName))));
```
We currently retrieve items from `ProcessCache` using the unique `processDefinitionKey`.
Therefore, a `Map` with the `processDefinitionKey` as the key and the corresponding process definition values wrapped in `ProcessCacheItem` seem to me the most appropriate data structure.
This object can be extended in the future if needed.

konstantinos-mylosis replied:
No, it is actually every processDefinition's FlowNode(id, name)
- process definition might  have more than one user task
- `ProcessCacheItem` is also used at `ElementInstanceServices` where all elements are needed

konstantinos-mylosis committed:
fix: refactor for cleaner code

use helper methods utility, predicate
rename methods
enhance tests with proper processName value

eppdot commented on service/src/main/java/io/camunda/service/cache/ProcessDefinitionProvider.java:
```diff
@@ -60,33 +66,46 @@ public void extractElementNames(
+        LOG.warn(
+            "Could not load all required process definitions into the cache. Expected: {}, Found: {}",
+            processDefinitionKeys.size(),
+            searchResult.total());
```
❤️

eppdot reviewed (changes requested):
Looks very good now 👍 . One simple change request

eppdot reviewed (approved):
🚀 nice work @konstantinos-mylosis

konstantinos-mylosis commented:
/backport backport stable/8.8
